modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/ToJenaConverters.scala

```diff
@@ -0,0 +1,71 @@
+package ch.epfl.bluebrain.nexus.rdf.jena
+
+import ch.epfl.bluebrain.nexus.rdf.Node.Literal.LanguageTag
+import ch.epfl.bluebrain.nexus.rdf.Node.{BNode, IriNode, IriOrBNode, Literal}
+import ch.epfl.bluebrain.nexus.rdf.{Graph, Node}
+import org.apache.jena.datatypes.TypeMapper
+import org.apache.jena.rdf.model
+import org.apache.jena.rdf.model._
+import org.apache.jena.rdf.model.impl.ResourceImpl
+
+import scala.util.Try
+
+/**
+  * Conversions from rdf to Jena to data types.
+  */
+trait ToJenaConverters {
+
+  /**
+    * Converts the argument `node` to a Jena `RDFNode`.
+    */
+  def asJena(node: Node): RDFNode = node match {
+    case ibn: IriOrBNode => asJena(ibn)
+    case lit: Literal    => asJena(lit)
+  }
+
+  /**
+    * Converts the argument `node` to a Jena `Resource`.
+    */
+  def asJena(node: IriOrBNode): Resource = node match {
+    case bn: BNode   => asJena(bn)
+    case in: IriNode => asJena(in)
+  }
+
+  /**
+    * Converts the argument `literal` to a Jena `Literal`.
+    */
+  def asJena(literal: Literal): model.Literal = literal match {
+    case Literal(lf, _, Some(LanguageTag(value))) => ResourceFactory.createLangLiteral(lf, value)
+    case Literal(lf, dt, _) =>
+      Try {
+        val tpe     = TypeMapper.getInstance().getSafeTypeByName(dt.asString)
+        val literal = ResourceFactory.createTypedLiteral(lf, tpe)
+        literal.getValue // It will throw whenever the literal does not match the desired datatype
+        literal
+      }.getOrElse(ResourceFactory.createStringLiteral(lf))
```

Comment:
Are there tests for this? I'm not able to find them

Reply:
These are tested through the syntax to cover both the syntax and the logic:
```
"node is a literal" in {
  val cases = List(
    ResourceFactory.createLangLiteral("a", "en-US")                 -> Literal("a", LanguageTag("en-US").rightValue),
    ResourceFactory.createStringLiteral("a")                        -> Literal("a"),
    ResourceFactory.createTypedLiteral("1", XSDDatatype.XSDinteger) -> Literal(1),
    ResourceFactory.createTypedLiteral("1.0", XSDDatatype.XSDfloat) -> Literal(1f),
    ResourceFactory.createLangLiteral("1", "en")                    -> Literal("1", LanguageTag("en").rightValue)
  )
  forAll(cases) {
    case (jl, l) =>
      (l: Node).asJena shouldEqual jl
      l.asJena shouldEqual jl
  }
}
```